proposals/extension_get_contexts.md

```diff
@@ -0,0 +1,171 @@
+types, and windows. It would be defined as:
+
+```
+extension.ContextProperties = {
+  tabId? = int,                // Find context by tab id, omitted returns all tabs.
+  contextType? = ContextType,  // Find context by type, omitted returns all types.
+  windowId? = int,             // Find context by window id, omitted returns all
+                               // windows.
+}
+```
+
+`ContextType` would serve a similar purpose of
+<code>[fetchProperties](https://developer.chrome.com/docs/extensions/reference/extension/#method-getViews)</code>’s
+[ViewType](https://developer.chrome.com/docs/extensions/reference/extension/#type-ViewType)
+in order to limit the type of extension contexts returned by the API. It would
+be defined as:
+
+```
+extension.ContextType = {
+  TAB: 'TAB',                               // Tabs the extension is running in.
+  POPUP: 'POPUP',                           // Popups created by the extension.
+  SERVICE_WORKER: 'SERVICE_WORKER',         // Service workers for the extension.
+  OFFSCREEN_DOCUMENT: 'OFFSCREEN_DOCUMENT'  // Offscreen documents for the extension.
+};
+```
+
+It would return an array of new `Context` objects for each frame the extension
+is running in:
+
+```
+extension.Context = {
+  tabId: int,                    // ID of the tab for this view.
+  windowId: int,                 // ID of the window this tab is located in.
+  documentId: int,               // ID of the DOM document loaded in the tab.
+  frameId: int,                  // ID of the frame this context is running for.
+  contextType: ContextType,      // Context type -- tab, popup, etc.
+  url: string,                   // The current URL of the tab.
+  origin: string,                // The origin of the context. Mostly this is the
+                                 // extension origin.
+  incognito: boolean             // Whether the context is for an incognito
+                                 // profile.
+}
+```
+
+`Context` objects will contain an `incognito` value indicating if the Context is
+running in an incognito profile. In this area, this brings up the case of an
+extension specifying
+[split incognito mode](https://developer.chrome.com/docs/extensions/mv3/manifest/incognito/#split).
+In this case we would not return the <code>Context</code>s from the related (but
+different) extension process running for the incognito profile.
+<code>spanning</code> mode would return any <code>Context</code>s running in an
+incognito profile (with <code>Context.incognito</code> being <code>true</code>)
+in the same shared process. More comments about this are in the Future Work
+section on this proposal.
+
+## Future Work
+
+### Multi Page Architecture Fields (`DocumentLifeCycle`, `FrameType`, and `parentDocumentId`, and `parentFrameId`)
+
+[Multi Page Architecture](http://doc/1NginQ8k0w3znuwTiJ5qjYmBKgZDekvEPC22q0I4swxQ#heading=h.w1qo2n6sr8wn)
+changed how outermost frames are presented in web navigation in chromium. More
```

Comment:
Chromium is a proper noun. The impact of MPArch wasn't limited to the Web Navigation API; it also affected the Tabs API and may have ripple effects on other platform features (e.g. documentId support in the Scripting API).

```suggestion
changed how outermost frames are exposed to extensions in Chromium. More
```